src/api/firebase.js

```diff
@@ -92,16 +93,77 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 export async function updateItem(listId, itemId, checked) {
 	const itemRef = doc(db, listId, itemId);
 	const itemDoc = await getDoc(itemRef);
-	const { dateLastPurchased, totalPurchases } = itemDoc.data();
+	const { dateCreated, dateLastPurchased, dateNextPurchased, totalPurchases } =
+		itemDoc.data();
 
-	try {
-		return updateDoc(itemRef, {
-			dateLastPurchased: checked ? new Date() : dateLastPurchased,
-			totalPurchases: checked ? totalPurchases + 1 : totalPurchases,
-			checked: checked,
-		});
-	} catch (error) {
-		console.log('updateDoc error', error);
+	const checkedDateLastPurchased = checked ? new Date() : dateLastPurchased;
+
+	if (checkedDateLastPurchased) {
+		// Convert dateNextPurchased from Timestamp to a JS Date
+		const dateNextPurchasedAsDate = dateNextPurchased.toDate();
+
+		/**
+		 * Calculate the previous estimate for purchase
+		 * based on historical data
+		 */
+		const previousEstimate = getDaysBetweenDates(
+			checkedDateLastPurchased,
+			dateNextPurchasedAsDate,
+		);
+
+		/**
+		 * Calculate the days since the last transaction,
+		 * considering either last purchased or the created date
+		 */
+		const daysSinceLastTransaction = getDaysBetweenDates(
+			new Date(),
+			checkedDateLastPurchased || dateCreated,
+		);
+
+		// Calculate the remaining days until the next purchase
+		let remainingDays = calculateEstimate(
+			previousEstimate,
+			daysSinceLastTransaction,
+			totalPurchases,
+		);
+
+		// Ensure that the next purchase date is at least one day in the future
+		if (remainingDays <= 0) {
+			remainingDays = 1;
+		}
+
```

Comment:
Yes, this was needed! 
One small suggestion: turn this magic number into a constant at the top of the function or the file:
const MINIMUM_REMAINING_DAYS = 1;

Reply:
I'm not a fan of magic numbers, but I don't know if this a case of a magic number. It seems readable as is, and the comments clarify the intent.

Reply:
Hi @CorinaMurg, thanks for bringing this up! The issue was that sometimes the `remainingDays` value would become 0, causing `dateNextPurchased` to be set for the same day as the purchase date, resulting in it becoming unchecked.

To address this (while retaining the "24 hours have passed" feature that Shangguan and I added last week), we introduced an additional check to ensure that `dateNextPurchased` is always *at least* one day ahead of the purchase date.

As @mentalcaries mentioned, this wouldn't necessarily be a magic number - we believe the comments adequately explain our intentions, eliminating the need to create a constant variable.

Reply:
Actually, @mentalcaries, regarding this, I'm still confused about the expected result of `dateNextPurchased` and whether we are achieving what was requested. If `dateNextPurchased` is set for 2 days from now, should the app automatically mark the item as "bought" when that date/time arrives? 🤔

Reply:
@rdpfeifle No it won't. As the coming issues will show, it has more to do with when the app recommends that something is due to be purchased. So if it's calculated to be 2 days from now, it might be given higher priority over something that's due to be purchased in 10 days. Hope this makes sense!

Reply:
@mentalcaries Ok, I think that makes sense. Thanks a lot! ✨

Reply:
@mentalcaries Ok, I think that makes sense. Thank you! 🙏🏼

Reply:
The const MINIMUM_REMAINING_DAYS = 1 suggestion was more in the spirit of the discussion from our last meeting re: cleaning up the comments to let the code flow better. The line of code would be self-explanatory and not rely on comments so much.